pkg/trait/affinity.go

```diff
@@ -129,16 +127,11 @@ func (t *affinityTrait) addNodeAffinity(_ *Environment, deployment *appsv1.Deplo
 		},
 	}
 
-	if deployment.Spec.Template.Spec.Affinity == nil {
```

Comment:
I wonder what has changed that avoids to have that nil check?

Reply:
I've refactored the 3 `add` methods and that check is done in the `Apply` method:
```
if podSpec.Affinity == nil {
			podSpec.Affinity = &corev1.Affinity{}
		}
```

Reply:
Ah right, I missed it 🤦🏼. Thanks.